Wangscape/OptionsManager.cpp

```diff
@@ -0,0 +1,54 @@
+#include "OptionsManager.h"
+
+#include <fstream>
+
+#include <boost/filesystem.hpp>
+
+#include <spotify/json.hpp>
+
+#include "codecs/OptionsCodec.h"
+
+OptionsManager::OptionsManager(std::string filename)
+{
+    std::ifstream ifs(filename);
+    if (!ifs.good())
+    {
+        throw std::runtime_error("Could not open options file");
+    }
+
+    std::string str{std::istreambuf_iterator<char>(ifs),
+                    std::istreambuf_iterator<char>()};
+
+    mOptions = spotify::json::decode<Options>(str.c_str());
+    mOptions.filename = filename;
+
+    // TODO(hryniuk): move it elsewhere
+    auto outputDirectory = mOptions.outputDirectory;
```

Comment:
Wangscape can generate a lot of output files and overwrite existing files, so it shouldn't be allowed to automatically choose an unexpected directory when it fails to open the user's choice of directory.

Prompting the user for an alternative output dir would be OK (but maybe unnecessary, if the fault is detected quickly).

Using SFML to display previews during or after tileset generation is a nice idea for the future.